src/frontend/circom/mod.rs

```diff
@@ -0,0 +1,182 @@
+fn num_bigint_to_ark_bigint(value: &num_bigint::BigInt) -> Result<ark_ff::BigInt<4>, Box<dyn Error>> {
+    let big_uint = value.to_biguint().ok_or_else(|| Box::new(std::io::Error::new(std::io::ErrorKind::Other, "BigInt is negative")))?;
+    Ok(ark_ff::BigInt::<4>::try_from(big_uint).map_err(|_| Box::new(std::io::Error::new(std::io::ErrorKind::Other, "BigInt conversion failed")))?)
+}
+
+// Convert R1CS constraints and witness from Circom format to folding-schemes R1CS and z format.
+pub fn circom_to_folding_schemes_r1cs_and_z(
+    constraints: Vec<Constraints<Bn254>>,
+    witness: &Vec<BigInt>,
+    public_inputs: usize,
+) -> Result<(R1CS<Fr>, Vec<Fr>), Box<dyn Error>> {
+    let folding_schemes_r1cs = convert_to_folding_schemes_r1cs(constraints, public_inputs);
+
+    let z: Vec<Fr> = witness
+        .iter()
+        .filter_map(|big_int| {
+            match num_bigint_to_ark_bigint(big_int) {
+                Ok(ark_big_int) => Fr::from_bigint(ark_big_int.into()),
+                Err(_) => None
+            }
+        })
+        .collect();
+
+    Ok((folding_schemes_r1cs, z))
+}
+
+#[cfg(test)]
+mod tests {
+    use ark_bn254::Bn254;
+    use num_bigint::BigInt;
+
+    use super::Constraints;
+    use super::calculate_witness;
+    use super::circom_to_folding_schemes_r1cs_and_z;
+    use super::convert_constraints_bigint_to_scalar;
+    use super::extract_constraints_from_r1cs;
+
+    #[test]
+    fn test_circom_to_folding_conversion() {
+        let current_dir = std::env::current_dir().unwrap();
+        
+        let base_path = current_dir.join("src/frontend/circom/test_folder");
+        let r1cs_filepath = base_path.join("test_circuit.r1cs");
+        let wasm_filepath = base_path.join("test_circuit.wasm");
+
+        /*
+        This is the test_circuit.cirom file.
+        When step_in equals 3, it will pass the test function.
+
+        template Example () {
+            signal input step_in;
+            signal output step_out;   
+            signal temp;
+            
+            temp <== step_in * step_in;
+            step_out <== temp * step_in + step_in + 5;
+            step_out === 35; 
+        }
+        component main = Example();
+        */
+    
+        assert!(r1cs_filepath.exists());
+        assert!(wasm_filepath.exists());
+    
+        let (constraints, public_inputs) = extract_constraints_from_r1cs(&r1cs_filepath).expect("Error");
+        assert!(!constraints.is_empty());
+    
+        let converted_constraints: Vec<Constraints<Bn254>> = constraints
+            .iter()
+            .map(|constraint| convert_constraints_bigint_to_scalar(constraint.clone()))
+            .collect();
+        assert_eq!(constraints.len(), converted_constraints.len());
+    
+        let inputs = vec![
+            // success
+            ("step_in".to_string(), vec![BigInt::from(3)]),
+            // fail
+            // ("step_in".to_string(), vec![BigInt::from(6)]),
```

Comment:
How do you see having a test or part of the test for this value instead of having the test that expects it to fail commented? in this way that part of the test ensures that it fails as expected.

Reply:
Thank you for the feedback! I changed the test functions have been segregated into success and failure.